sandy/decay.py

```diff
@@ -197,12 +200,9 @@ def custom_perturbation_decayconstant(self, pert):
                     'branching_ratio': 7.2e-11,
                     'branching_ratio_uncertainty': 2.1e-11}}}}
         """
-        dc = self.get_decayconstant().LAMBDA
-        dc = dc * (1 + pert)
-        pert_dc = self.data
-        for key in self.data:
-            pert_dc[key]['decay_constant'] = dc[key]
-        return self.__class__(pert_dc)
+        pert_dc = self.data[zam].copy()
+        pert_dc['decay_constant'] = pert_dc['decay_constant'] * (1 + pert)
+        return self.__class__({zam: pert_dc})

```

Comment:
I like this better...but here you return in the DecayData output only the ZAM to which you appiedato the perturbation, while the others are lost.

For instance, in the example you lose the info on U238.
Is this the behavior you want? I think it makes more sense to keep all nuclides....what do you think?

Reply:
I thought that the output could only show the zam with the updated decay constant but you are right, it is better to report all the zam otherwise to see and use them I have to recall the input file. I fixed it and also imported the copy module to use deepcopy to edit only the copy